saul-examples/src/test/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/EntityRelation/EntityRelationTests.scala

```diff
@@ -69,4 +70,37 @@ class EntityRelationTests extends FlatSpec with Matchers {
     val results = PersonClassifier.crossValidation(5)
     results.foreach { case score => (score.overall.f1 > minScore) should be(true) }
   }
+  "Initialization on ER " should "work." in {
+
+    EntityRelationDataModel.clearInstances()
+    EntityRelationDataModel.populateWithConllSmallSet()
+
+    val cls_base= List(PersonClassifier,OrganizationClassifier,LocationClassifier,WorksForClassifier, LivesInClassifier)
+    val cls = List(PerConstrainedClassifier, OrgConstrainedClassifier, LocConstrainedClassifier, WorksFor_PerOrg_ConstrainedClassifier, LivesIn_PerOrg_relationConstrainedClassifier)
+
+    ClassifierUtils.ForgetAll(cls_base:_*)
+
+    PerConstrainedClassifier.onClassifier.classifier.getLabelLexicon.size() should be (0)
+    PerConstrainedClassifier.onClassifier.classifier.getLexicon.size() should be (0)
+
+    ClassifierUtils.InitializeClassifiers(pairs,cls:_*)
+
+    PerConstrainedClassifier.onClassifier.classifier.getLabelLexicon.size() should be (2)
+    PerConstrainedClassifier.onClassifier.classifier.getLexicon.size() should be (84)
+
+    ClassifierUtils.TrainClassifiers(5,cls_base)
+
+    PerConstrainedClassifier.onClassifier.classifier.asInstanceOf[SparseNetworkLearner].getNetwork.get(0).asInstanceOf[LinearThresholdUnit].getWeightVector.size() should be (1661)
+
+    val jointTrainIteration = 5
+    JointTrainSparseNetwork.train[ConllRelation](
+    pairs, cls, jointTrainIteration, init = false)
+
+    PerConstrainedClassifier.onClassifier.classifier.asInstanceOf[SparseNetworkLearner].getNetwork.get(0).asInstanceOf[LinearThresholdUnit].getWeightVector.size() should be (1661)
+
```

Comment:
Okey, I see. The problem is that I ca not have the multiple apply with initialization. So I changed it and we need to pass a flag always then.